internal/server/subscription_manager.go

```diff
@@ -200,12 +195,11 @@ func (c *subscriptionManagerCore) UnsubscribeAll() {
 }
 
 func (c *subscriptionManagerCore) subscribe(name string) {
-	_, ok := c.subscriptions[name]
-	if !ok {
-		c.subscriptions[name] = c.locator.Subscribe(c.ctx, c.typeURL, name, c.handler)
-	} else {
-		c.locator.Resubscribe(c.ctx, c.typeURL, name, c.handler)
+	unsub, ok := c.subscriptions[name]
```

Comment:
Can we not call `c.unsubscribe` here? It seems almost identical except is also does a `delete(c.subscriptions, name)`, which seems safe to do here as `c.subscriptions[name]` is set immediately after

Reply:
Sure, makes sense. Technically, it's an additional unneeded map lookup for the `delete`, but I really doubt that'll matter :)